### DISCUSSION TITLE
NameIt plugin only generates warnings, no text

### DISCUSSION THREAD

**Discussion by CZKR** [5axes/NameIt#18] [Q&A] [answered]
After trying for 30 minutes to get the NameIt plugin to work, figuring that this plugin might work similar to the TabAntiWarp plugin via the side controls (nope) because that's the first 5axes plugin I installed, and then reading through the GitHub wiki information and still getting no results, I decided I must be doing something wrong and searched a demonstration video.  The only one I found was from Willi's 3D-Druck (German) Youtube channel https://www.youtube.com/watch?v=SGea_PNpEtc

After watching and duplicating the video's steps, all I get is "Warning! NameIT ... No identifier created for 0 element(s)".

The Identification mesh is checked on, but even after repeated Cura restarts, different models loaded, it's  the same result.  The ONLY thing that does seem to work for me so far is the Rename Models function.  All other NameIt functions are not producing anything.

BTW thanks so much for your other plugins - they're so helpful!

Using Cura 5.2.1

**Comment by 5axes** [comment]
If you use the function menu "Add Name" whithout any selection and if you have some parts on the build plate . It should work. A Project file will be welcome to analyse your issue

**Comment by CZKR** [comment]
Hi, I believe I figured out what was going wrong (bug?).  After several tests loading different .stl models, leaving them un-sliced or slicing, and switching different profiles in Cura (eg. normal, draft, and my long named custom ones, it turns out that the NameIT plugin will always fail if the user is currently using a profile with a "#" in the name.  This is actually fairly common to happen too since Cura will automatically add the "#" symbol into newly created profiles when the user does a "Create profile from current settings" action - something I do all the time when tweaking settings for new models.  I create new profiles with long names all the time.

As soon as I tried "Add Name" using a profile without "#" in the title, it worked immediately.

I even found a temporary "workaround" in that I can setup one or more models on the build plate using whatever profile (including ones with "#" in the title) to get the settings just right), then at the end when I'm ready to add names (which is usually the last action you would want to do before slicing), switch to say the NORMAL profile, add the name(s) so they appear on the build plate too.  Then finally switch back to your desired profile that might have "#" in the title, and slice.  Once the NameIt routine has done it's job, and the names are already on the build plate, it doesn't care if you switch profiles afterwards.

**Reply by 5axes** [reply]
Need to be tested, I don"t see the link between the profile name and this plugin

**Comment by 5axes** [comment]
As expected no issue for me concerning the name of the profile : 
![image](https://user-images.githubusercontent.com/11015345/210329812-9741d15d-6220-4521-bbb0-79fd44f53187.png)
  

Without an example project ( 3mf file ) it will be hard to analyse your issue.

**Comment by CZKR** [comment]
[NameItTroubleshooting.zip](https://github.com/5axes/NameIt/files/10348955/NameItTroubleshooting.zip)

OK here's a simple test cube 3mf file for you inside the zip file (fyi - I used 7zip).  I confirmed that if I select profile "CJ E3S1Pro PLA Figurines Non-Tree #2" it does fail.  If I choose "CJ E3S1Pro Draft #3" or another shorter titled profile, it does appear work properly.  So my initial thought about the "#" character was incorrect.  It now appears to matter how long the profile name is as to whether it will work or not.

**Comment by 5axes** [comment]
The good News : I can now reproduce your issue : 
![image](https://user-images.githubusercontent.com/11015345/210723815-c0c435c1-03c7-45d9-9ab5-b590a0267b92.png)

The bad news .. It's not a question of profile name length , Even with a longer name I don't have any issue. 
![image](https://user-images.githubusercontent.com/11015345/210724787-f83f1c80-c8da-411e-982a-c7eb6573dfcf.png)

So as it is not so obvious, it could take time to analyse the issue origine.

**Comment by 5axes** [comment, marked as answer]
Got It !!!!

To identify the element I have created a new settings call Identification Mesh .... In your profile the Identification Mesh Is checked. So every element as considered as an existing identification mesh . 
![image](https://user-images.githubusercontent.com/11015345/210730537-a9980b3e-c799-4928-9e21-05c6d0d39f57.png)

You Need to uncheck this parameter in your Profile.

**Comment by 5axes** [comment]
![image](https://user-images.githubusercontent.com/11015345/210731073-6f0c29eb-2614-49f0-85e4-b04ec6e171db.png)
